galette/lib/Galette/Core/Plugins.php

```diff
@@ -375,6 +376,27 @@ public function loadSmarties($id)
         }
     }
 
+    /**
+     * Loads event provider
+     *
+     * @param string $id Module ID
+     *
+     * @return void
+     */
+    public function loadEventProviders($id, $namespace)
+    {
+        global $emitter;
+
+        $f = $this->modules[$id]['root'] . '/lib/' . $namespace. '/PluginEventProvider.php';
+        if (file_exists($f)) {
+            $providerClassName = '\\' . $namespace . '\\' . 'PluginEventProvider';
+            if (class_exists($providerClassName) &&
+                method_exists($providerClassName, 'provideListeners')) {
+                $emitter->useListenerProvider(new $providerClassName());
+            }
+        }
```

Comment:
Les plugins utilisent normalement le mécanisme d'autoload de Galette, mais pourraient fournir le leur aussi ; autant ne vérifier que la présence de la classe je pense.

Donc, à mon avis, quelque chose comme ceci serait davantage adapté :
```suggestion
        $providerClassName = '\\' . $namespace . '\\' . 'PluginEventProvider';
        if (class_exists($providerClassName) &&
            method_exists($providerClassName, 'provideListeners')) {
            $emitter->useListenerProvider(new $providerClassName());
        }
```

Qu'en penses-tu @nbehier ?

Reply:
C'est effectivement mieux !